File: src/Java/uk/co/armedpineapple/cth/NavDrawerAdapter.java
package uk.co.armedpineapple.cth;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

public class NavDrawerAdapter extends BaseAdapter {

    final Context         context;
    final List<MenuItems> items;

    public NavDrawerAdapter(Context context, List<MenuItems> items) {
        this.context = context;
        this.items = items;
    }

    @Override
    public int getCount() {
        return items.size();
    }

    @Override
    public Object getItem(int position) {
        return items.get(position);
    }

    @Override
    public long getItemId(int position) {
		return position;
	}

	@Override
	public View getView(int position, View convertView, ViewGroup parent) {

		View view;
		MenuItems mi = (MenuItems) getItem(position);

		if (convertView != null) {
			view = convertView;
		} else {
			LayoutInflater inflater = (LayoutInflater) context
					.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
			view = inflater.inflate(R.layout.menu_item, parent, false);
		}
		TextView tv = (TextView) view.findViewById(R.id.menu_item_text);
		tv.setText(mi.textResource);

		return view;
	}

}

File: src/Java/uk/co/armedpineapple/cth/NavDrawerAdapter.kt
package uk.co.armedpineapple.cth

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import kotlinx.android.synthetic.main.menu_item.view.*

class NavDrawerAdapter(internal val context: Context, internal val items: List<MenuItems>) : BaseAdapter() {

    private val inflater: LayoutInflater by lazy {
        context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun getCount(): Int {
        return items.size
    }

    override fun getItem(position: Int): MenuItems {
        return items[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val view: View = convertView
                ?: inflater.inflate(R.layout.menu_item, parent, false)

        val mi = getItem(position)

        if (mi.textResource != null) {
            view.menuItemTextView.setText(mi.textResource)
        }

        return view
    }

}
